### PR TITLE
(PCP-452) Add option to configure Association timeout

### PR DESCRIPTION
**Pull request by parisiale**
The `association-timeout` option is now used to set the timeout for the
overall Association process (default to 15 s).

The old `association-timeout` (ttl for the Association request) can now
be specified by the `association-request-ttl` option (default to 10 s).

Renaming the `pcp-message-timeout` option to 'pcp-messag-ttl', for
consistency with the other ttl option.

### CODE REVIEW

**Comment by parisiale** [comment]
Requires https://github.com/puppetlabs/cpp-pcp-client/pull/180.

File: lib/src/configuration.cc [hunk 1]
```diff
@@ -818,9 +830,13 @@ void Configuration::validateAndNormalizeOtherSettings()
         throw Configuration::Error {
             lth_loc::translate("association-timeout must be positive") };
 
-    if (HW::GetFlag<int>("pcp-message-timeout") < 0)
+    if (HW::GetFlag<int>("association-request-ttl") < 0)
+        throw Configuration::Error {
+                lth_loc::translate("association-request-ttl must be positive") };
```
Comment (mruzicka) [inline lib/src/configuration.cc]: Is zero legal? The test does allow it.

File: lib/src/configuration.cc [hunk 2]
```diff
@@ -818,9 +830,13 @@ void Configuration::validateAndNormalizeOtherSettings()
...
+
+    if (HW::GetFlag<int>("pcp-message-ttl") < 0)
         throw Configuration::Error {
-            lth_loc::translate("pcp-message-timeout must be positive") };
+            lth_loc::translate("association-request-ttl must be positive") };
```
Comment (mruzicka) [inline lib/src/configuration.cc]: Same as above.

Comment (parisiale) [inline lib/src/configuration.cc]: If you configure 0, you should know what you're doing.

**Comment by mruzicka** [comment]
LGTM

**Pull request merged**